python/triton/language/extra/hip/libdevice.py

```diff
@@ -0,0 +1,438 @@
+from triton.language import core
+
+
+@core.builtin
+@core._tensor_member_fn
+def abs(x, _builder=None):
+    x = core._to_tensor(x, _builder)
+    if x.dtype.is_floating():
+        return core.tensor(_builder.create_fabs(x.handle), x.type)
+    elif x.dtype.is_int_signed():
+        return core.tensor(_builder.create_iabs(x.handle), x.type)
+    else:
+        assert False, f"Unexpected dtype {x.dtype}"
+
+
+@core.builtin
+@core._tensor_member_fn
+def floor(x, _builder=None):
+    x = core._to_tensor(x, _builder)
+    if x.dtype.is_floating():
+      return core.tensor(_builder.create_floor(x.handle), x.type)
+    else:
+      assert False, f"Unexpected dtype {x.dtype}"
+
+
+@core.extern
+def rsqrt(arg0, _builder=None):
+    return core.extern_elementwise(
+        "", "", [arg0], {
+            (core.dtype("fp32"), ): ("__ocml_rsqrt_f32", core.dtype("fp32")),
+            (core.dtype("fp64"), ): ("__ocml_rsqrt_f64", core.dtype("fp64")),
+        }, is_pure=True, _builder=_builder)
+
+
+@core.extern
+def ceil(arg0, _builder=None):
+    return core.extern_elementwise(
+        "", "", [arg0], {
+            (core.dtype("fp64"), ): ("__ocml_ceil_f64", core.dtype("fp64")),
+            (core.dtype("fp32"), ): ("__ocml_ceil_f32", core.dtype("fp32")),
+        }, is_pure=True, _builder=_builder)
+
+
+@core.extern
+def trunc(arg0, _builder=None):
+    return core.extern_elementwise(
+        "", "", [arg0], {
+            (core.dtype("fp64"), ): ("__ocml_trunc_f64", core.dtype("fp64")),
+            (core.dtype("fp32"), ): ("__ocml_trunc_f32", core.dtype("fp32")),
+        }, is_pure=True, _builder=_builder)
+
+
+@core.extern
+def exp2(arg0, _builder=None):
+    return core.extern_elementwise(
+        "", "", [arg0], {
+            (core.dtype("fp32"), ): ("__ocml_exp2_f32", core.dtype("fp32")),
+            (core.dtype("fp64"), ): ("__ocml_exp2_f64", core.dtype("fp64")),
+        }, is_pure=True, _builder=_builder)
+
+
+@core.extern
+def sqrt(arg0, _builder=None):
+    return core.extern_elementwise(
+        "", "", [arg0], {
+            (core.dtype("fp32"), ): ("__ocml_sqrt_f32", core.dtype("fp32")),
+            (core.dtype("fp64"), ): ("__ocml_sqrt_f64", core.dtype("fp64")),
+        }, is_pure=True, _builder=_builder)
+
+
+# TODO: take from MLIR
+@core.extern
+def llrint(arg0, _builder=None):
+    return core.extern_elementwise(
+        "", "", [
+            arg0,
+        ], {
+            (core.dtype("fp32"), ): ("__nv_llrintf", core.dtype("int64")),
+            (core.dtype("fp64"), ): ("__nv_llrint", core.dtype("int64")),
```

Comment:
if we don't have a function with int64 input in ocml, maybe we can convert the input to float or double and call ocml_rint_f32/f64?